### PR TITLE
Feat/401 update screen five to add tested parent tree pct on frontend

### PR DESCRIPTION
**Pull request by craigyu**
# Description
Implements #401 

#### Front end
- Reduced the spaghetti-ness of `index.tsx` in ParentTreeStep, moved some functions to utils
- Implemented the info sections according to the new design
- Integrated the gw calc endpoint to display values when users click on the calculate values button
##### New Type:
- `src/types/JsxChildren.ts`: A type that allows passing children to components, see the usage in `InfoSection`
##### New Util:
- `src/utils/RecordUtils.ts`: Provides 3 utils functions similar to `Object.keys()`, `Object.values()` and `Object.entities()`. Devs can use these to get the keys and values from a `Record` object.

#### Back end
- Renamed a couple variables
- Handled an edge case where the algorithm could divide values by zero

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Local testing

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-438-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-438-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-438-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-438-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-438-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-438-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

**Comment by craigyu** [comment]
image
🙂🔫

**Comment by craigyu** [comment]
Close to reopen

File: frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/utils.ts [hunk 1]
```diff
@@ -80,3 +91,307 @@ export const calcSum = (tableRows: Array<RowItem>, field: string): string => {
...
+      invalidParentTreeNumbers.push(parentTreeNumber);
+    }
+  });
+
+  setStepData(clonedState);
+
+  if (invalidParentTreeNumbers.length > 0) {
+    // A temporary solution to let users know they have invalid clone numbers
+    // eslint-disable-next-line no-alert
+    alert(`The following clone numbers cannot be found: ${invalidParentTreeNumbers}`);
```
Comment (RMCampos) [inline frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/utils.ts:253]: 👍🏻

**Review by RMCampos — APPROVED** [review]
Great job! Monster! 🥇 💯

**Pull request merged**